src/spatial/detail/ArborX_Iota.hpp

```diff
@@ -0,0 +1,54 @@
+/****************************************************************************
+ * Copyright (c) 2024 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_IOTA_HPP
+#define ARBORX_IOTA_HPP
+
+#include <detail/ArborX_AccessTraits.hpp>
+
+#include <Kokkos_Macros.hpp>
+
+#include <type_traits>
+
+namespace ArborX::Experimental
+{
+
+template <typename MemorySpace, typename Index = int>
+struct Iota
+{
+  using memory_space = MemorySpace;
```

Comment:
```suggestion
template <typename Index = int>
struct Iota
{
```

Reply:
It needs `MemorySpace` so that we can use it as `Values` constructor argument, and to wrap into `AccessValues`.